AY2021S1-CS2103T-F11-4/tp
# Fix last modified field not updating on mutable updates

## Description

`LastModifiedInstant` was not updated on client note adding/deleting/editing since they were mutable updates. This PR updates the field.

## Testing

Run `client suggest by/frequency`

1. Edit a client's field that is not at index 1. E.g. `client edit 2 n/New Name` -> Client should now be at top of list
1. Add a note to a client that is not at index 1. E.g. `client note add 2 nt/New Note` -> Client should now be at top of list
1. Edit a note to a client that is not at index 1. E.g. `client note edit 2 1 n/Edit Note` -> Client should now be at top of list
1. Delete a note to a client that is not at index 1. E.g. `client note delete 2 1` -> Client should now be at top of list
